src/session/mpi.rs

```diff
@@ -125,28 +139,40 @@ impl SessionMPI {
 
         for provider in &self.provider_sessions {
             provider
-                .download(blob_id, "app".to_owned(), ResourceFormat::Tar)
+                .download(blob_id, app_path.clone(), ResourceFormat::Tar)
                 .context("downloading file")?;
 
+            // If we create a ProviderSession per provider, every session
```

Comment:
Where is the symlink created?

Reply:
It a part of the image: https://github.com/golemfactory/gumpi-image/blob/master/tmp